test/commandbuilderharness.js

```diff
@@ -0,0 +1,161 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+const weiroll = require("@weiroll/weiroll.js");
+
+async function deployLibrary(name) {
+  const factory = await ethers.getContractFactory(name);
+  const contract = await factory.deploy();
+  return weiroll.Contract.fromEthersContract(contract);
+}
+
+describe("CommandBuilderHarness", function () {
+  let cbh;
+  let math;
+  let strings;
+  let abi = ethers.utils.defaultAbiCoder;
+
+  before(async () => {
+    const Cbh = await ethers.getContractFactory("CommandBuilderHarness");
+    cbh = await Cbh.deploy();
+
+    math = await deployLibrary("Math");
+    strings = await deployLibrary("Strings");
+  });
+
+  async function executeBuildInputs(commands, state, abiout, msg){
+    for (let c of commands) {
+        selector = c.slice(0, 10);
+        indices = "0x" + c.slice(10, 10+7*2);
+        target = "0x" + c.slice(10+7*2);
+        const tx = await cbh.testBuildInputs(state, selector, indices);
+        await expect(tx)
+          .to.emit(cbh, "BuiltInput")
+          .withArgs(selector + abiout.slice(2));
+        const receipt = await tx.wait();
+        console.log(`buildInputs for ${msg} : ${receipt.gasUsed.toNumber()} gas`);
+    }
+  }
+
+  it("Should build inputs that match Math.add ABI", async () => {
+    const planner = new weiroll.Planner();
+
+    let args = [1, 2];
+
+    abiout = abi.encode(math.interface.getFunction("add").inputs, args);
+
+    planner.addCommand(math.add.apply(this, args));
+    planner.addCommand(math.add.apply(this, args));
```

Comment:
Any reason to be doing this twice?

Reply:
Ah I think that's a remnant from when I was testing something else, I'll remove